crates/nu-cli/src/commands/commandline/edit.rs

```diff
@@ -58,9 +60,15 @@ impl Command for SubCommand {
             repl.buffer.insert_str(cursor_pos, &str);
             repl.cursor_pos += str.len();
         } else {
+            // default == "replace"
             repl.buffer = str;
             repl.cursor_pos = repl.buffer.len();
         }
+        if call.has_flag(engine_state, stack, "accept")? {
+            if let Ok(mut queue) = engine_state.reedline_event_queue.lock() {
+                queue.push(ReedlineEvent::Enter);
```

Comment:
`ReedlineEvent::Enter` is not solitary confined to running the command.

- if a menu is active it will perform its accepting action instead
- it triggers a run of the `!` history expression expansions if necessary
- it will run the syntax validation to decide if it should insert a newline due to open expressions
- if not it will return the entered content

https://github.com/nushell/reedline/blob/020142f6ee857353c289119a04d8fc71c474ce59/src/engine.rs#L1085-L1120
so this implementation choice may have surprising behavior based on the state of the line editor (see https://github.com/nushell/reedline/pull/821 for folks wanting to have an always open menu)

Reply:
I see, what do you suggest? `Submit`?

Reply:
Ooh and where do I find the code that does the history expansion? I would also give it a shot to use this feature to instantly accept a `!!` expansion.